shell/browser/api/electron_api_push_notifications_mac.mm

```diff
@@ -19,28 +19,23 @@
   gin_helper::Promise<std::string> promise(isolate);
   v8::Local<v8::Promise> handle = promise.GetHandle();
 
-  [[AtomApplication sharedApplication]
-      registerForRemoteNotificationTypes:NSRemoteNotificationTypeBadge |
-                                         NSRemoteNotificationTypeAlert |
-                                         NSRemoteNotificationTypeSound];
+  [[AtomApplication sharedApplication] registerForRemoteNotifications];
 
   PushNotifications::apns_promise_set_.emplace_back(std::move(promise));
   return handle;
 }
 
 void PushNotifications::ResolveAPNSPromiseSetWithToken(
     const std::string& token_string) {
-  std::vector<gin_helper::Promise<std::string>> promises =
-      std::move(PushNotifications::apns_promise_set_);
+  auto promises = std::move(PushNotifications::apns_promise_set_);
```

Comment:
Not new to this PR & not a blocker here, but just as a heads-up ---  just noticed this code is relying on [UB](https://stackoverflow.com/questions/17730689/is-a-moved-from-vector-always-empty) of what the state of `apns_promise_set_` is after it's been `move()`d. tldr is it's tricky to `move()` a field that stays around & can potentially be referenced again after the move.

We probably do this elsewhere in our code, too. I'll take a look